src/lamp_py/aws/s3.py

```diff
@@ -181,6 +181,21 @@ def object_metadata(obj: str) -> Dict[str, str]:
         raise error
 
 
+def version_check(obj: str, version: str) -> bool:
+    """
+    compare an s3 file's lamp version to a given version
+
+    :return True if remote and expected version match, else False
+    """
+    try:
+        remote_version = object_metadata(obj).get("lamp_version", "")
+
+        return remote_version == version
+
+    except Exception:
+        return False
+
+
```

Comment:
new utility function. i think its best to keep the key for our versions only in this file going forward to avoid typos.

should probably update the upload file signature as well.